.travis.yml

```diff
@@ -1,36 +1,34 @@
----
 dist: xenial
-
 language: python
-
 python:
-  - "3.6"
-
+- '3.6'
 services:
-  - docker
-
+- docker
 env:
-  - mysql_root_username=root mysql_root_password='12345'
-
+  global:
+  - mysql_root_username=root
+  - mysql_root_password='12345'
+  - mysql_host=localhost
+  - mysql_db=devopsloft
 before_install:
-  - sudo apt-get install -y libmysqlclient-dev
-
+- sudo apt-get install -y libmysqlclient-dev
+- cp configuration.yml web_s2i/configuration.yml
 addons:
   apt:
     update: true
-
-cache:
-  apt: true`
-
 install:
-  - pip install -r requirements-dev.txt
-  - docker build -t devopsloft/devopsloft docker/.
-
+- pip install -r tests/requirements-test.txt
+- docker build -t devopsloft/mysql db_s2i/.
+- docker build -t devopsloft/devopsloft web_s2i/.
 before_script:
-  - docker run -d -p 80:80 -p 3306:3306 --name web devopsloft/devopsloft
-  - docker exec -it web bash -c 'while ! mysqladmin ping --silent; do sleep 3; done'
-
+- docker network create devopsloft_network
+- docker run -d --network devopsloft_network -p 3306:3306 --name db -e MYSQL_ROOT_PASSWORD=12345 -e MYSQL_DATABASE=devopsloft devopsloft/mysql
+- docker logs db
+- docker run -d --network devopsloft_network -p 80:80 --name web devopsloft/devopsloft
+- docker logs web
+- docker exec -it db timeout 60 bash -c 'while ! mysqladmin ping --silent; do sleep 3; done'
 script:
-  - flake8 --exclude venv .
-  - tests/is_table_empty.py
-  - tests/basic_http_get_test.py
+- flake8 --exclude venv .
+- sleep 10
```

Comment:
why the sleep is required?

Reply:
A workaround for:
"mysql.connector.errors.OperationalError: 2013 (HY000): Lost connection to MySQL server at 'reading initial communication packet', system error: 0"